src/main/java/appeng/me/storage/BasicCellInventory.java

```diff
@@ -23,50 +6,90 @@
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
 
-import appeng.api.AEApi;
 import appeng.api.config.Actionable;
 import appeng.api.exceptions.AppEngException;
 import appeng.api.implementations.items.IStorageCell;
 import appeng.api.networking.security.IActionSource;
-import appeng.api.storage.ICellInventoryHandler;
-import appeng.api.storage.IMEInventory;
+import appeng.api.storage.ICellInventory;
 import appeng.api.storage.ISaveProvider;
 import appeng.api.storage.IStorageChannel;
-import appeng.api.storage.channels.IItemStorageChannel;
 import appeng.api.storage.data.IAEItemStack;
+import appeng.api.storage.data.IAEStack;
 import appeng.core.AEConfig;
 import appeng.core.AELog;
+import appeng.util.item.AEStack;
 
 
-public class ItemCellInventory extends AbstractCellInventory<IAEItemStack>
+public class BasicCellInventory<T extends IAEStack<T>> extends AbstractCellInventory<T>
 {
-	private ItemCellInventory( final ItemStack o, final ISaveProvider container ) throws AppEngException
+	private final IStorageChannel<T> channel;
+
+	private BasicCellInventory( final IStorageCell<T> cellType, final ItemStack o, final ISaveProvider container )
 	{
-		super( o, container, 8 );
+		super( cellType, o, container );
+		this.channel = cellType.getChannel();
 	}
 
-	public static ICellInventoryHandler<IAEItemStack> getCell( final ItemStack o, final ISaveProvider container2 )
+	public static <T extends IAEStack<T>> ICellInventory<T> createInventory( final ItemStack o, final ISaveProvider container )
 	{
 		try
 		{
-			return new ItemCellInventoryHandler( new ItemCellInventory( o, container2 ) );
+			if( o == null )
+			{
+				throw new AppEngException( "ItemStack was used as a cell, but was not a cell!" );
+			}
+
+			final Item type = o.getItem();
+			final IStorageCell<T> cellType;
+			if( type instanceof IStorageCell )
+			{
+				cellType = (IStorageCell<T>) type;
+			}
+			else
+			{
+				throw new AppEngException( "ItemStack was used as a cell, but was not a cell!" );
+			}
+
+			if( !cellType.isStorageCell( o ) )
+			{
+				throw new AppEngException( "ItemStack was used as a cell, but was not a cell!" );
+			}
+
+			return new BasicCellInventory<T>( cellType, o, container );
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2293)